components/service/glean/src/main/java/mozilla/components/service/glean/CommonMetricData.kt

```diff
@@ -76,4 +79,25 @@ interface CommonMetricData {
         val filteredNames = sendInPings.filter { it != DEFAULT_STORAGE_NAME }
         return filteredNames + defaultStorageDestinations
     }
+
+    /**
+     * Helper function to help await Jobs returned from coroutine launch executions used by metrics
+     * when recording data.  This is to help ensure that data is updated before test functions
+     * check or access them.
+     *
+     * @param job Job that is to be awaited
+     * @return true if job completed in the given time, otherwise false if it timed out
+     */
+    fun awaitJob(job: Job): Boolean {
+        var loopCnt = MAX_WAIT_LOOP_CNT
+        while (job.isActive && loopCnt > 0) {
+            --loopCnt
+            Thread.sleep(MAX_WAIT_SLEEP_DELAY)
+        }
```

Comment:
Since you already have a `Job`here. Can't you [join](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-job/join.html) on that? You may need to wrap that in [runBlocking](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/run-blocking.html). Is that something that would work here?

Reply:
I initially looked at doing this but was nervous about somehow blocking indefinitely with `join()`, which is why I went with a polled solution that had an out after a reasonable amount of time.

Reply:
You may be able to work around that by using [withTimeout](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/with-timeout.html), I think.

Reply:
You may be able to work around that by using [withTimeout](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/with-timeout.html), I think.

Reply:
Thanks!  I like this much better now 👍

Reply:
Yeah.  It feels like at least if something isn't right it's not our problem ;)

Reply:
This looks good!